File: ruby/bob/bob.rb
# frozen_string_literal: true

class Bob
  def self.hey(remark)
    case remark.strip
    when /^\s*$/
      'Fine. Be that way!'
    when /\A[A-Z\s]+\?\Z/
      "Calm down, I know what I'm doing!"
    when /([A-Z\s]+\Z)|([A-Z \d]+!+\Z)/
      'Whoa, chill out!'
    when /.*\?\Z/
      'Sure.'
    else
      'Whatever.'
    end
  end
end

class BookKeeping
  VERSION = 2
end

File: python/bob/bob.py
import re

def hey(phrase):
    phrase = phrase.rstrip()

    if re.match('^\s*$', phrase):
        return 'Fine. Be that way!'
    elif re.match('[A-Z\s]*\?\Z', phrase):
        return "Calm down, I know what I'm doing!"
    elif re.match('.*\?\Z', phrase):
        return 'Sure.'
    elif re.match('[A-Z\s]+\Z', phrase) or re.match('[A-Z \W\d]+!+\Z', phrase):
        return 'Whoa, chill out!'
    else:
        return 'Whatever.'
